spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/dsl/MongoDbTests.java

```diff
@@ -0,0 +1,351 @@
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.mongodb.MongoClient;
+
+/**
+ * @author Xavier Padró
+ * @since 5.0
+ */
+@RunWith(SpringRunner.class)
+@DirtiesContext
+public class MongoDbTests extends MongoDbAvailableTests {
+	private static final String COLLECTION_NAME = "data";
+
+	@Autowired
+	private PollableChannel getResultChannel;
+
+	@Autowired
+	@Qualifier("gatewaySingleQueryFlow.input")
+	private MessageChannel gatewaySingleQueryFlow;
+
+	@Autowired
+	@Qualifier("gatewaySingleQueryWithTemplateFlow.input")
+	private MessageChannel gatewaySingleQueryWithTemplateFlow;
+
+	@Autowired
+	@Qualifier("gatewaySingleQueryExpressionFlow.input")
+	private MessageChannel gatewaySingleQueryExpressionFlow;
+
+	@Autowired
+	@Qualifier("gatewayQueryExpressionFlow.input")
+	private MessageChannel gatewayQueryExpressionFlow;
+
+	@Autowired
+	@Qualifier("gatewayQueryExpressionLimitFlow.input")
+	private MessageChannel gatewayQueryExpressionLimitFlow;
+
+	@Autowired
+	@Qualifier("gatewayQueryFunctionFlow.input")
+	private MessageChannel gatewayQueryFunctionFlow;
+
+	@Autowired
+	@Qualifier("gatewayCollectionNameFunctionFlow.input")
+	private MessageChannel gatewayCollectionNameFunctionFlow;
+
+	@Autowired
+	private MongoOperations mongoTemplate;
+
+	@Before
+	public void setUp() throws Exception {
+		createPersons();
+	}
+
+	@After
+	public void cleanUp() {
+		mongoTemplate.dropCollection(COLLECTION_NAME);
+	}
+
+	@Test
+	@MongoDbAvailable
+	public void testGatewayWithSingleQuery() {
+		gatewaySingleQueryFlow.send(MessageBuilder.withPayload("Xavi").build());
+
+		Message<?> result = this.getResultChannel.receive(10_000);
+
+		assertNotNull(result);
+		Person retrievedPerson = (Person) result.getPayload();
+		assertEquals("Xavi", retrievedPerson.getName());
+	}
+
+	@Test
+	@MongoDbAvailable
+	public void testGatewayWithSingleQueryWithTemplate() {
+		gatewaySingleQueryWithTemplateFlow.send(MessageBuilder.withPayload("Xavi").build());
+
+		Message<?> result = this.getResultChannel.receive(10_000);
+
+		assertNotNull(result);
+		Person retrievedPerson = (Person) result.getPayload();
+		assertEquals("Xavi", retrievedPerson.getName());
+	}
+
+	@Test
+	@MongoDbAvailable
+	public void testGatewayWithSingleQueryExpression() {
+		gatewaySingleQueryExpressionFlow.send(MessageBuilder
+				.withPayload("")
+				.setHeader("query", "{'name' : 'Artem'}")
+				.build());
+
+		Message<?> result = this.getResultChannel.receive(10_000);
+
+		assertNotNull(result);
+		Person retrievedPerson = (Person) result.getPayload();
+		assertEquals("Artem", retrievedPerson.getName());
+	}
+
+	@Test(expected = ReplyRequiredException.class)
```

Comment:
M-m-m, do we expose `requires-reply` attribute to the XML config?
```
<xsd:attribute name="requires-reply" type="xsd:string" use="optional" default="true">
	<xsd:annotation>
		<xsd:documentation>
			Specify whether this outbound gateway must return a non-null value. This value is
			'true' by default, and a ReplyRequiredException will be thrown when
			the underlying service returns a null value.
		</xsd:documentation>
	</xsd:annotation>
</xsd:attribute>
```
The Java DSL should be fine for that: `ConsumerEndpointSpec.requiresReply()`